scripts/jenkins/ardana/gerrit/build_test_package.py

```diff
@@ -434,42 +312,37 @@ def build_test_packages(change_ids, obs_linked_project, home_project,
     # 'gerrit_project': Package()
     packages = {}
 
+    # The target branch associated with the first change is used for
+    # all changes
+    branch = None
+
     # Grab each change for the supplied change_ids. As we go through the
     # changes the change_ids list may expand with dependencies. These are also
     # processed. If a change has already been processed we skip it to avoid
     # circular dependencies.
-    try_branches = ['master']
     for id in change_ids:
-        if id in processed_changes:
-            # Duplicate dependency, skipping..
-            continue
-        c = GerritChange(id, try_branches)
-        # id might not have been in canonical form
+        c = GerritChange(id, branch=branch)
         if c.id in processed_changes:
-            processed_changes.append(id)
             # Duplicate dependency, skipping..
             continue
-        processed_changes.append(id)
         processed_changes.append(c.id)
-
-        if c.target not in try_branches:
-            # Save the branch as a hint to disambiguate cross-repo
-            # dependencies.
-            try_branches.insert(0, c.target)
+        branch = branch or c.branch
 
         # skip packages that don't have asssociated RPMs
         if c.gerrit_project not in gerrit_project_map():
             print("Warning: Project %s has no RPM, Skipping"
                   % c.gerrit_project)
 
+            # Add the dependent changes to the change_ids to process
+            change_ids.extend(find_dependency_headers(c.commit_message))
```

Comment:
We should instead use `c.load_dependencies()` so that the c.dependencies() points to the same objects. We can then recursively place c.dependencies() into the packages to build.

Reply:
Oops, looks like I missed this. Let me correct that.

Reply:
Ok, done. I also switched to using a list of GerritChange objects, which makes this even easier.